lib/src/Pos/Generator/Block/Mode.hs

```diff
@@ -375,6 +379,19 @@ instance MonadBlockGenBase m => MonadBListener (BlockGenMode ext m) where
 instance Monad m => MonadAddresses (BlockGenMode ext m) where
     type AddrData (BlockGenMode ext m) = Address
     getNewAddress = pure
+    getFakeChangeAddress = pure (go 0)
+      where
+        -- We want it to be deterministic. It must be consistent with
+        -- the way we construct address in block-gen. If it's changed,
+        -- tests will fail, so we will notice it.
+        go :: Word8 -> Address
+        go i =
+            let addr =
+                    makePubKeyAddressBoot . fst $
```

Comment:
Why do you consider only boot addresses here?

Reply:
Because currently block-gen only uses `makePubKeyAddressBoot` to generate addresses.
It's a bit fragile to rely on it here, but, as I wrote in the comment above, if it stops being true, tests will fail and we will notice it immediately.

Reply:
Alright, this makes sense then.